cpp/ql/src/semmle/code/cpp/security/Encryption.qll

```diff
@@ -42,17 +54,48 @@ string algorithmWhitelist() {
   result = "ECIES"
 }
 
-/** A regex for matching strings that look like they contain a whitelisted algorithm */
-string algorithmWhitelistRegex() {
-  // The implementation of this is a duplicate of algorithmBlacklistRegex, as it isn't
-  // possible to have string -> string functions at the moment
+/**
+ * Returns a regular expression for matching strings that look like they
```

Comment:
```suggestion
 * Gets a regular expression for matching strings that look like they
```

I wonder why this has multiple results while its sibling predicate concatenates everything into one super-regex. Perhaps the predicate should be renamed to `getASecureAlgorithmRegex` (added `A`).

Reply:
I hadn't noticed that.  I'd be more inclined to change it to concatenate them?

Reply:
Concatenating them would be better, but perhaps that should be a follow-up so we can get this merged today?

Reply:
Sounds good to me.

Reply:
https://github.com/github/codeql/pull/3856